<div style="text-align: center;">
  <img src="https://github.com/Hack-io-Data/Imagenes/blob/main/01-LogosHackio/logo_naranja@4x.png?raw=true" alt="esquema" />
</div>

# Laboratorio Pandas (EDA y Unión de Datos)

## Los datos

Para este laboratorio, trabajarás con los siguientes conjuntos de datos:

- **`netflix_originals.csv`**: Contiene información sobre producciones originales de Netflix. Las columnas que nos encontraremos en este conjunto de datos son: 

   - `Title`: Contiene el nombre del título de la producción original de Netflix.

   - `Genre`: Especifica el género o combinación de géneros de la producción.

   - `Premiere`: Indica la fecha de estreno de la producción en Netflix.

   - `Runtime`: Contiene la duración de la producción en minutos.

   - `IMDB Score`: Representa la calificación promedio que la producción ha recibido en la plataforma IMDB.

   - `Language`: Especifica el idioma principal o los idiomas en los que se presenta la producción.

- **`netflix_titles.csv`**: Contiene información sobre una variedad de títulos disponibles en la plataforma Netflix, incluyendo tanto películas como series de televisión. Las columnas que nos encontraremos en este conjunto de datos son:

   - `show_id`: Identificador único para cada título en el dataset.

   - `type`: Especifica el tipo de contenido, ya sea "Movie" (película) o "TV Show" (serie de televisión).

   - `title`: Contiene el nombre del título de la producción.

   - `director`: Muestra el nombre del director del título. Puede estar vacío en algunos casos.

   - `cast`: Lista de actores y actrices que participaron en el título. También puede estar vacío en algunos casos.

   - `country`: País o países donde se produjo el título. Esta columna también puede tener valores faltantes.

   - `date_added`: Fecha en la que el título fue añadido al catálogo de Netflix.

   - `release_year`: Año en que el título fue lanzado o producido.

   - `rating`: Clasificación del contenido en términos de audiencia (por ejemplo, PG-13, TV-MA).

   - `duration`: Duración del título. Para películas, se expresa en minutos (e.g., "90 min") y para series en temporadas (e.g., "2 Seasons").

   - `listed_in`: Categorías o géneros bajo los cuales el título está clasificado (por ejemplo, "Documentaries", "TV Dramas").

   - `description`: Breve sinopsis o descripción del contenido del título.

A lo largo de los laboratorios de Pandas, trabajarás con estos conjuntos de datos (o con los derivados del Laboratorio), algunas de las actividades que realizaremos son:

- Exploración de datos y análisis exploratorio (EDA) para familiarizarse con los conjuntos de datos.

- Realización de uniones entre conjuntos de datos, aplicando diferentes técnicas de *merging* y analizando la relevancia de cada método.

- Uso de funciones de selección y filtrado (`loc`, `iloc`) para extraer información clave.

- Desarrollo de análisis descriptivos y visualización de patrones para comprender mejor las decisiones estratégicas de la plataforma.

----

## Objetivo General de la Serie de Laboratorios



1. **Unión de `netflix_titles.csv` con `netflix_originals.csv`:**

   - **Análisis de contenido original:**  Identificarás los títulos que son producciones originales de Netflix y los compararás con los títulos adquiridos. Este análisis te permitirá:

     - Comparar la diversidad de géneros entre los títulos originales y no originales.

     - Evaluar si ciertos géneros o tipos de contenido (como documentales o series) son más frecuentes en las producciones originales.

     - Investigar si la producción original se concentra en determinados países o idiomas.


   - **Evaluación de contenido:** En esta etapa, explorarás si las producciones originales de Netflix tienden a recibir mejores o peores evaluaciones en comparación con el contenido adquirido. Algunas preguntas que se buscarán responder incluyen:

     - ¿Los títulos originales tienen una calificación promedio superior en plataformas como IMDB?

     - ¿Son las producciones originales más consistentes en términos de calidad percibida?

     - ¿Existen diferencias en la popularidad de los títulos originales según su género o país de origen?


   - **Identificación de tendencias:**  Por último, este análisis te permitirá explorar cómo ha evolucionado la estrategia de contenido de Netflix a lo largo del tiempo. Algunos puntos clave a investigar serán:

     - ¿Cómo ha crecido la producción original en comparación con la adquisición de contenido?

     - ¿Qué géneros o tipos de contenido han recibido mayor inversión en los últimos años?
     
     - ¿Existen patrones en los países o idiomas de las producciones originales, especialmente en relación con la expansión global de Netflix?


----


## Ejercicio: Análisis y Enriquecimiento del Catálogo de Netflix

### Parte 1: Análisis Exploratorio de Datos (EDA)

1. **Carga de los datos:**

   - Deberás cargar los dos conjuntos de datos que te hemos proporcionado para realizar este Laboratorio:

     - `netflix_originals.csv`

     - `netflix_titles.csv`

   Deberás revisar la estructura de cada conjunto de datos utilizando los métodos aprendidos durante la lección.

2. **Análisis de columnas y consistencia:**

   - Deberás hacer un análisis exploratorio de cada uno de los conjuntos de datos. Será obligatorio que incluyas las conclusiones generales obtenidas de cada uno de los análisis exploratorios en el Jupyter Notebook de tus soluciones.

   - Explorar qué columnas contiene cada conjunto de datos y cuáles son sus tipos de datos. Identificar si hay valores nulos o duplicados.

   - Algunas preguntas que nos podemos plantear en esta fase son (recuerda que esto es solo un ejemplo, y que podemos explorar todo lo que queramos/necesitemos para entender lo mejor posible los conjuntos de datos):

      - ¿Cuáles son los tipos de datos de cada columna? Nos puede ayudar a detectar si algún tipo de dato está incorrectamente asignado, como fechas almacenadas como cadenas de texto.

      - ¿Existen valores nulos? ¿En qué columnas están? Este paso es crucial para decidir si deben ser eliminados, rellenados o ignorados.

      - ¿Cuántos títulos únicos hay en cada conjunto de datos? Nos puede permitir verificar si existen títulos duplicados que puedan distorsionar el análisis.

      - ¿Qué géneros tenemos en el conjunto de datos? Esto nos puede ayudar a identificar los géneros más populares o incluso a identificar una posible limpieza de dicha columna para poder extraer conclusiones más claras en nuestro análisis.

### Parte 2: Unión de Datos

1. **Realización de la unión:**

   - Deberás realizar la unión que consideres más apropiada entre `netflix_titles.csv` y `netflix_originals.csv`. Además, deberás justificar el tipo de unión que has realizado.

   - Esto nos ayudará a contestar preguntas más adelente del tipo:

     - ¿Cuántos títulos en `netflix_titles.csv` son producciones originales de Netflix?

     - ¿Hay diferencias significativas en la puntuación de IMDB entre títulos originales y no originales?

### Parte 3: Uso de `loc` e `iloc`

Usando el DataFrame creado en la Parte 2: 

1. **Filtrado de datos con `loc`:**

   - Selecciona todas las filas donde el tipo de contenido sea “Movie”.

   - Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.

   - Selecciona todas las filas donde el país sea “United States”.

   - Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.

   - Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.

   - Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).

2. **Acceso a datos específicos con `iloc`:**

   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.

   - Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.

   - Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.


**NOTA IMPORTANTE**: Antes de terminar el lab, guarda el conjunto de datos creado en la fase de unión para trabajar con el en las siguientes lecciones. 

In [2]:
import pandas as pd
import numpy as np

# Parte 1

**DATOS ORIGINALS**

In [3]:
df_noriginals = pd.read_csv("datos/netflix_originals.csv")
df_noriginals.head()

Unnamed: 0            Title                  Genre           Premiere  \
0           0  Enter the Anime            Documentary     August 5, 2019   
1           1      Dark Forces               Thriller    August 21, 2020   
2           2          The App  Science fiction/Drama  December 26, 2019   
3           3   The Open House        Horror thriller   January 19, 2018   
4           4      Kaali Khuhi                Mystery   October 30, 2020   

   Runtime  IMDB Score          Language  
0       58         2.5  English/Japanese  
1       81         2.6           Spanish  
2       79         2.6           Italian  
3       94         3.2           English  
4       90         3.4             Hindi

Esta es un amuestra de 5 filas del contenido del csv netflix_originals

In [4]:
df_noriginals.shape

(513, 7)

Gracias a .shape, podemos observar que tenemos 513 filas en este archivo y un total de 7 columnas, 6 al eliminar la columna Unnamed: 0

In [5]:
df_noriginals.drop(columns = ["Unnamed: 0"], inplace = True)
df_noriginals.head()

Title                  Genre           Premiere  Runtime  \
0  Enter the Anime            Documentary     August 5, 2019       58   
1      Dark Forces               Thriller    August 21, 2020       81   
2          The App  Science fiction/Drama  December 26, 2019       79   
3   The Open House        Horror thriller   January 19, 2018       94   
4      Kaali Khuhi                Mystery   October 30, 2020       90   

   IMDB Score          Language  
0         2.5  English/Japanese  
1         2.6           Spanish  
2         2.6           Italian  
3         3.2           English  
4         3.4             Hindi

Quitamos la columna Unnamed: 0, ya que nos es ineccesaria y nos ensucia los datos

In [6]:
display(df_noriginals.tail())
display(df_noriginals.info())

Title         Genre  \
508                       Cuba and the Cameraman   Documentary   
509                       Dancing with the Birds   Documentary   
510  Winter on Fire: Ukraine's Fight for Freedom   Documentary   
511                      Springsteen on Broadway  One-man show   
512     David Attenborough: A Life on Our Planet   Documentary   

              Premiere  Runtime  IMDB Score                  Language  
508  November 24, 2017      114         8.3                   English  
509   October 23, 2019       51         8.3                   English  
510    October 9, 2015       91         8.4  English/Ukranian/Russian  
511  December 16, 2018      153         8.5                   English  
512    October 4, 2020       83         9.0                   English

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 24.2+ KB


None

Utilizo .tail para darme cuenta un poco de como puede ser mi conjunto de datos, ya que .head solo nos muestra 5 filas.  
También uso info para darme cuenta que tipo de dato almacena cada columna
- Es por ello que me puedo dar cuenta que que la columna Title , Genere, Premiere y Lenguage el tipo de dantro que contiene son objects, Runtime son Integers y IMDB Score utiliza float
- Podemos observar también que no hay datos de tipo null en este conjunto de datos

In [7]:
df_noriginals["Title"].value_counts(ascending=False)

Title
David Attenborough: A Life on Our Planet    1
Shirkers                                    1
The Black Godfather                         1
The Irishman: In Conversation               1
The Speed Cubers                            1
                                           ..
Kaali Khuhi                                 1
Drive                                       1
Leyla Everlasting                           1
The Last Days of American Crime             1
Paradox                                     1
Name: count, Length: 513, dtype: int64

Podemos ver que en el primer conjunto de datos no hay titulos repetidos

In [8]:
df_noriginals["Genre"].value_counts()

Genre
Documentary                             132
Drama                                    73
Comedy                                   42
Romantic comedy                          35
Thriller                                 33
                                       ... 
Action-adventure                          1
Concert Film                              1
Animation / Science Fiction               1
Anthology/Dark comedy                     1
Animation/Christmas/Comedy/Adventure      1
Name: count, Length: 106, dtype: int64

Observamos la cantidad de Peliculas que hay por Genero

In [9]:
df_noriginals["Language"].value_counts()

Language
English                       352
Hindi                          28
Spanish                        26
French                         18
Italian                        14
Portuguese                      9
Indonesian                      9
Japanese                        5
English/Spanish                 5
Korean                          5
Turkish                         4
German                          4
Marathi                         3
Polish                          3
English/Japanese                2
Dutch                           2
Filipino                        2
Thai                            2
English/Hindi                   2
Norwegian                       1
Malay                           1
Swedish                         1
Spanish/Basque                  1
English/Swedish                 1
Spanish/Catalan                 1
English/Taiwanese/Mandarin      1
Thia/English                    1
Bengali                         1
Khmer/English/French            1
Tamil

Observamos la cantidad de peliculas que hay por Idioma  
Podemos observar que hay un numero de peliculas que tienen idioma doble, es decir en su version original hablan en diferentes idiomas.

In [10]:
df_noriginals["Title"].is_unique

True

La funcion .is_true devuelve True si todos los datos son unicos y False si hay algun dato repetido, en este caso no hay datos repetidos

**DATOS TITLES**

In [11]:
df_ntitles = pd.read_csv("datos/netflix_titles.csv")
df_ntitles.head()

Unnamed: 0 show_id     type                  title         director  \
0           0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1           1      s2  TV Show          Blood & Water              NaN   
2           2      s3  TV Show              Ganglands  Julien Leclercq   
3           3      s4  TV Show  Jailbirds New Orleans              NaN   
4           4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [12]:
df_ntitles.drop(columns = ["Unnamed: 0"], inplace = True)
df_ntitles.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

De momento llevamos un proceso similar al anterior conjunto de datos eliminando al Unnamed: 0

In [13]:
df_ntitles.shape

(8807, 12)

Nos fijamos en la forma de nuestro conjunto de datos

In [14]:
df_ntitles.info()
df_ntitles.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration        4813
listed_in          0
description        0
dtype: int64

Gracias a .info nos podemos fijar que todas las columnas usan el tipo de dato Object, excepto release_year que es int64.  
Nos podemos fijar que en las columnas de director, cast, country, date_added, rating y duration contienen valores Null.  
Gracias a .isnull y a poder sumarlos, como podemos ver mas en detalla cuantas entradas Null hay en cada columna

In [15]:
df_ntitles["director"].value_counts()

director
Rajiv Chilaka              19
Raúl Campos, Jan Suter     18
Marcus Raboy               16
Suhas Kadav                16
Jay Karas                  14
                           ..
Mu Chu                      1
Chandra Prakash Dwivedi     1
Majid Al Ansari             1
Peter Hewitt                1
Paul Raschid                1
Name: count, Length: 4528, dtype: int64

Observamos el numero de titulos que ha dirigido cada director

In [16]:
df_ntitles["title"].is_unique

True

Todos los valores son unicos

# Parte 2

In [17]:
titulos_tabla_originales = df_noriginals["Title"].unique()
display(titulos_tabla_originales)
lista_titulos_coincidentes = []
for i in titulos_tabla_originales:
    if i in df_ntitles["title"].unique():
        print(i)
        lista_titulos_coincidentes.append(i)
        


array(['Enter the Anime', 'Dark Forces', 'The App', 'The Open House',
       'Kaali Khuhi', 'Drive', 'Leyla Everlasting',
       'The Last Days of American Crime', 'Paradox', 'Sardar Ka Grandson',
       'The Call', 'Whipped', 'All Because of You', 'Mercy',
       'After the Raid', 'Ghost Stories', 'The Last Thing He Wanted',
       'What Happened to Mr. Cha?', 'Death Note', 'Secret Obsession',
       'Sextuplets', 'The Girl on the Train', 'Thunder Force',
       'Fatal Affair', 'Just Say Yes', 'Seriously Single',
       'The Misadventures of Hedi and Cokeman', '5 Star Christmas',
       'After Maria', 'I Am the Pretty Thing That Lives in the House',
       'Paris Is Us', 'Rattlesnake', 'The Players', 'We Are One',
       'Finding Agnes', 'IO', 'Sentinelle', 'Sol Levante', 'The Binding',
       'We Can Be Heroes', 'Christmas Crossfire', 'Coin Heist',
       'Mrs. Serial Killer', 'Nobody Sleeps in the Woods Tonight',
       'Take the 10', 'The Main Event', 'The Ridiculous 6',
       'Ea

Enter the Anime
Dark Forces
The App
The Open House
Kaali Khuhi
Drive
Leyla Everlasting
The Last Days of American Crime
Paradox
Sardar Ka Grandson
The Call
Whipped
All Because of You
Mercy
After the Raid
Ghost Stories
The Last Thing He Wanted
What Happened to Mr. Cha?
Death Note
Secret Obsession
Sextuplets
The Girl on the Train
Thunder Force
Fatal Affair
Just Say Yes
Seriously Single
The Misadventures of Hedi and Cokeman
5 Star Christmas
After Maria
I Am the Pretty Thing That Lives in the House
Paris Is Us
Rattlesnake
The Players
We Are One
Finding Agnes
IO
Sentinelle
Sol Levante
The Binding
We Can Be Heroes
Christmas Crossfire
Coin Heist
Mrs. Serial Killer
Nobody Sleeps in the Woods Tonight
Take the 10
The Main Event
The Ridiculous 6
Earth and Blood
Fearless
Holiday Rush
The Day of the Lord
Airplane Mode
How It Ends
Love Like the Falling Rain
Rebirth
Squared Love
Cadaver
Clinical
Coffee & Kareem
Dude
Geez & Ann
The Larva Island Movie
American Factory: A Conversation with the Obamas
Des

en la variable lista_titulos_coincidentes, aparecen todos los titulos que SI son originales en el dataframe de Titles.

In [18]:
df_concat = df_ntitles.merge(df_noriginals, how="inner", right_on="Title", left_on="title")
display(df_concat)

show_id   type                                             title  \
0        s1  Movie                              Dick Johnson Is Dead   
1      s142  Movie                                        Extraction   
2      s625  Movie                                        Prime Time   
3      s835  Movie                                      Blue Miracle   
4      s837  Movie                                         Ghost Lab   
..      ...    ...                                               ...   
508   s5902  Movie                                               Tig   
509   s5907  Movie                       What Happened, Miss Simone?   
510   s5911  Movie                                  Hot Girls Wanted   
511   s5913  Movie  The Other One: The Long Strange Trip of Bob Weir   
512   s5917  Movie                                        My Own Man   

                          director  \
0                  Kirsten Johnson   
1                 Steven C. Miller   
2                     Jakub Piątek   
3                   Julio Quintana   
4              Paween Purijitpanya   
..                             ...   
508  Kristina Goolsby, Ashley York   
509                     Liz Garbus   
510       Jill Bauer, Ronna Gradus   
511                    Mike Fleiss   
512                David Sampliner   

                                                  cast  \
0                                                  NaN   
1    Bruce Willis, Kellan Lutz, Gina Carano, D.B. S...   
2    Bartosz Bielenia, Magdalena Popławska, Andrzej...   
3    Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...   
4    Thanapob Leeratanakachorn, Paris Intarakomalya...   
..                                                 ...   
508                                         Tig Notaro   
509                                                NaN   
510                                                NaN   
511                                           Bob Weir   
512                                    David Sampliner   

                                   country          date_added  release_year  \
0                            United States  September 25, 2021          2020   
1    United States, United Kingdom, Canada   September 1, 2021          2015   
2                                      NaN       June 30, 2021          2021   
3                            United States        May 27, 2021          2021   
4                                 Thailand        May 27, 2021          2021   
..                                     ...                 ...           ...   
508                          United States       July 17, 2015          2015   
509                          United States       June 26, 2015          2015   
510                          United States        May 29, 2015          2015   
511                          United States        May 22, 2015          2015   
512                          United States       March 6, 2015          2015   

    rating duration                                          listed_in  \
0    PG-13   90 min                                      Documentaries   
1        R   82 min                                 Action & Adventure   
2    TV-MA      NaN            Dramas, International Movies, Thrillers   
3    TV-PG      NaN  Children & Family Movies, Dramas, Faith & Spir...   
4    TV-MA  118 min     Horror Movies, International Movies, Thrillers   
..     ...      ...                                                ...   
508  TV-14      NaN                     Documentaries, Stand-Up Comedy   
509  TV-14      NaN                    Documentaries, Music & Musicals   
510  TV-MA   83 min                                      Documentaries   
511  TV-14   84 min                    Documentaries, Music & Musicals   
512  TV-MA   82 min                                      Documentaries   

                                           description  \
0    As her father nears the end of his life, filmm...   
1    When a retired

al usar un inner join, su comportamiento es que en el momento en el que no haya uno de los dos titulos en alguna de la lista, no lo anote, entonces fijandonos en la dimension de la tabla nueva, nos fijamos que son 513, los titulos que coincicen.

# Parte 3

## Parte 3.1 

In [19]:
df_concat.loc[df_concat["type"] == "Movie"]

show_id   type                                             title  \
0        s1  Movie                              Dick Johnson Is Dead   
1      s142  Movie                                        Extraction   
2      s625  Movie                                        Prime Time   
3      s835  Movie                                      Blue Miracle   
4      s837  Movie                                         Ghost Lab   
..      ...    ...                                               ...   
508   s5902  Movie                                               Tig   
509   s5907  Movie                       What Happened, Miss Simone?   
510   s5911  Movie                                  Hot Girls Wanted   
511   s5913  Movie  The Other One: The Long Strange Trip of Bob Weir   
512   s5917  Movie                                        My Own Man   

                          director  \
0                  Kirsten Johnson   
1                 Steven C. Miller   
2                     Jakub Piątek   
3                   Julio Quintana   
4              Paween Purijitpanya   
..                             ...   
508  Kristina Goolsby, Ashley York   
509                     Liz Garbus   
510       Jill Bauer, Ronna Gradus   
511                    Mike Fleiss   
512                David Sampliner   

                                                  cast  \
0                                                  NaN   
1    Bruce Willis, Kellan Lutz, Gina Carano, D.B. S...   
2    Bartosz Bielenia, Magdalena Popławska, Andrzej...   
3    Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...   
4    Thanapob Leeratanakachorn, Paris Intarakomalya...   
..                                                 ...   
508                                         Tig Notaro   
509                                                NaN   
510                                                NaN   
511                                           Bob Weir   
512                                    David Sampliner   

                                   country          date_added  release_year  \
0                            United States  September 25, 2021          2020   
1    United States, United Kingdom, Canada   September 1, 2021          2015   
2                                      NaN       June 30, 2021          2021   
3                            United States        May 27, 2021          2021   
4                                 Thailand        May 27, 2021          2021   
..                                     ...                 ...           ...   
508                          United States       July 17, 2015          2015   
509                          United States       June 26, 2015          2015   
510                          United States        May 29, 2015          2015   
511                          United States        May 22, 2015          2015   
512                          United States       March 6, 2015          2015   

    rating duration                                          listed_in  \
0    PG-13   90 min                                      Documentaries   
1        R   82 min                                 Action & Adventure   
2    TV-MA      NaN            Dramas, International Movies, Thrillers   
3    TV-PG      NaN  Children & Family Movies, Dramas, Faith & Spir...   
4    TV-MA  118 min     Horror Movies, International Movies, Thrillers   
..     ...      ...                                                ...   
508  TV-14      NaN                     Documentaries, Stand-Up Comedy   
509  TV-14      NaN                    Documentaries, Music & Musicals   
510  TV-MA   83 min                                      Documentaries   
511  TV-14   84 min                    Documentaries, Music & Musicals   
512  TV-MA   82 min                                      Documentaries   

                                           description  \
0    As her father nears the end of his life, filmm...   
1    When a retired

In [20]:
df_concat.loc[df_concat["type"] == "TV Show"]

Empty DataFrame
Columns: [show_id, type, title, director, cast, country, date_added, release_year, rating, duration, listed_in, description, Title, Genre, Premiere, Runtime, IMDB Score, Language]
Index: []

In [21]:
df_concat.loc[df_concat["country"] == "United States"]

show_id   type                                             title  \
0        s1  Movie                              Dick Johnson Is Dead   
3      s835  Movie                                      Blue Miracle   
7      s854  Movie                                  Army of the Dead   
11     s890  Movie                           The Woman in the Window   
14     s911  Movie                                           Monster   
..      ...    ...                                               ...   
508   s5902  Movie                                               Tig   
509   s5907  Movie                       What Happened, Miss Simone?   
510   s5911  Movie                                  Hot Girls Wanted   
511   s5913  Movie  The Other One: The Long Strange Trip of Bob Weir   
512   s5917  Movie                                        My Own Man   

                          director  \
0                  Kirsten Johnson   
3                   Julio Quintana   
7                      Zack Snyder   
11                      Joe Wright   
14                 Anthony Mandler   
..                             ...   
508  Kristina Goolsby, Ashley York   
509                     Liz Garbus   
510       Jill Bauer, Ronna Gradus   
511                    Mike Fleiss   
512                David Sampliner   

                                                  cast        country  \
0                                                  NaN  United States   
3    Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...  United States   
7    Dave Bautista, Ella Purnell, Omari Hardwick, G...  United States   
11   Amy Adams, Gary Oldman, Anthony Mackie, Fred H...  United States   
14   Kelvin Harrison Jr., Jennifer Hudson, Jeffrey ...  United States   
..                                                 ...            ...   
508                                         Tig Notaro  United States   
509                                                NaN  United States   
510                                                NaN  United States   
511                                           Bob Weir  United States   
512                                    David Sampliner  United States   

             date_added  release_year rating duration  \
0    September 25, 2021          2020  PG-13   90 min   
3          May 27, 2021          2021  TV-PG      NaN   
7          May 21, 2021          2021      R  148 min   
11         May 14, 2021          2021      R      NaN   
14          May 7, 2021          2021  TV-MA      NaN   
..                  ...           ...    ...      ...   
508       July 17, 2015          2015  TV-14      NaN   
509       June 26, 2015          2015  TV-14      NaN   
510        May 29, 2015          2015  TV-MA   83 min   
511        May 22, 2015          2015  TV-14   84 min   
512       March 6, 2015          2015  TV-MA   82 min   

                                             listed_in  \
0                                        Documentaries   
3    Children & Family Movies, Dramas, Faith & Spir...   
7                    Action & Adventure, Horror Movies   
11                                   Dramas, Thrillers   
14                                              Dramas   
..                                                 ...   
508                     Documentaries, Stand-Up Comedy   
509                    Documentaries, Music & Musicals   
510                                      Documentaries   
511                    Documentaries, Music & Musicals   
512                                      Documentaries   

                                           description  \
0    As her father nears the end of his life, filmm...   
3    To save their cash-strapped orphanage, a guard...   
7    After a zombie outbreak in Las Vegas, a group ...   
11   Confined to her home by agoraphobia, a psychol...   
14   A talented teen implicated in a robbery-turned...   
..                                                 ...   
508  Comedian Tig

In [22]:
df_concat.loc[df_concat["type"] == "Movie", ["title", "director"]]


title  \
0                                Dick Johnson Is Dead   
1                                          Extraction   
2                                          Prime Time   
3                                        Blue Miracle   
4                                           Ghost Lab   
..                                                ...   
508                                               Tig   
509                       What Happened, Miss Simone?   
510                                  Hot Girls Wanted   
511  The Other One: The Long Strange Trip of Bob Weir   
512                                        My Own Man   

                          director  
0                  Kirsten Johnson  
1                 Steven C. Miller  
2                     Jakub Piątek  
3                   Julio Quintana  
4              Paween Purijitpanya  
..                             ...  
508  Kristina Goolsby, Ashley York  
509                     Liz Garbus  
510       Jill Bauer, Ronna Gradus  
511                    Mike Fleiss  
512                David Sampliner  

[513 rows x 2 columns]

In [23]:
df_concat.loc[df_concat["release_year"] == 2018, ["title", "listed_in"]]


title  \
251             Little Miss Sumo   
320           Nappily Ever After   
324     Period. End of Sentence.   
331                         Soni   
334                    Lionheart   
..                           ...   
413      The Cloverfield Paradox   
414  A Futile and Stupid Gesture   
415                 Step Sisters   
416               The Open House   
417               The Polka King   

                                             listed_in  
251  Documentaries, International Movies, Sports Mo...  
320                  Comedies, Dramas, Romantic Movies  
324                                      Documentaries  
331   Dramas, Independent Movies, International Movies  
334               Comedies, Dramas, Independent Movies  
..                                                 ...  
413               Action & Adventure, Sci-Fi & Fantasy  
414                                           Comedies  
415                                           Comedies  
416                           Horror Movies, Thrillers  
417                       Comedies, Independent Movies  

[87 rows x 2 columns]

In [24]:
df_concat.loc[df_concat["director"].isna(), "title"]

24                     Dolly Parton: A MusiCares Tribute
30                                              Bad Trip
59                                           Stuck Apart
62                          The Minimalists: Less Is Now
70     Ma Rainey's Black Bottom: A Legacy Brought to ...
77                                     Leyla Everlasting
126             The Boys in the Band: Something Personal
142                                       Rising Phoenix
146                    John Was Trying to Contact Aliens
194             Murder to Mercy: The Cyntoia Brown Story
201                                         LA Originals
210                   Crip Camp: A Disability Revolution
234                        The Irishman: In Conversation
273     American Factory: A Conversation with the Obamas
278                       Invader Zim: Enter the Florpus
282                                            Otherhood
316            Antoine Griezmann: The Making of a Legend
329                            

## Parte 3.2

In [28]:
# df_concat.iloc[:5, [df_concat.columns.get_loc('title'), df_concat.columns.get_loc('director'), df_concat.columns.get_loc('country')]]
df_concat.iloc[:5, [2, 3, 5]]


title             director  \
0  Dick Johnson Is Dead      Kirsten Johnson   
1            Extraction     Steven C. Miller   
2            Prime Time         Jakub Piątek   
3          Blue Miracle       Julio Quintana   
4             Ghost Lab  Paween Purijitpanya   

                                 country  
0                          United States  
1  United States, United Kingdom, Canada  
2                                    NaN  
3                          United States  
4                               Thailand

In [32]:
df_concat.iloc[-5:, [2, 11]]


title  \
508                                               Tig   
509                       What Happened, Miss Simone?   
510                                  Hot Girls Wanted   
511  The Other One: The Long Strange Trip of Bob Weir   
512                                        My Own Man   

                                           description  
508  Comedian Tig Notaro announced her cancer diagn...  
509  Using never-before-heard recordings, rare arch...  
510  This 2015 Sundance Film Festival breakout docu...  
511  This chronicle of Bob Weir highlights his brot...  
512  When a man discovers he will be the father to ...

In [37]:
df_concat.iloc[:10, 1:6]


type                        title             director  \
0  Movie         Dick Johnson Is Dead      Kirsten Johnson   
1  Movie                   Extraction     Steven C. Miller   
2  Movie                   Prime Time         Jakub Piątek   
3  Movie                 Blue Miracle       Julio Quintana   
4  Movie                    Ghost Lab  Paween Purijitpanya   
5  Movie  Baggio: The Divine Ponytail    Letizia Lamartire   
6  Movie         Nail Bomber: Manhunt        Daniel Vernon   
7  Movie             Army of the Dead          Zack Snyder   
8  Movie           Sardar Ka Grandson        Kaashvie Nair   
9  Movie                        Ferry    Cecilia Verheyden   

                                                cast  \
0                                                NaN   
1  Bruce Willis, Kellan Lutz, Gina Carano, D.B. S...   
2  Bartosz Bielenia, Magdalena Popławska, Andrzej...   
3  Jimmy Gonzales, Dennis Quaid, Anthony Gonzalez...   
4  Thanapob Leeratanakachorn, Paris Intarakomalya...   
5  Andrea Arcangeli, Valentina Bellè, Andrea Penn...   
6                                                NaN   
7  Dave Bautista, Ella Purnell, Omari Hardwick, G...   
8  Arjun Kapoor, Neena Gupta, Rakul Preet Singh, ...   
9  Frank Lammers, Elise Schaap, Huub Stapel, Moni...   

                                 country  
0                          United States  
1  United States, United Kingdom, Canada  
2                                    NaN  
3                          United States  
4                               Thailand  
5                                  Italy  
6                                    NaN  
7                          United States  
8                                  India  
9                   Belgium, Netherlands

In [38]:
df_concat.iloc[-7:, [2, 3, 5]]


title  \
506       Winter on Fire: Ukraine's Fight for Freedom   
507               Keith Richards: Under the Influence   
508                                               Tig   
509                       What Happened, Miss Simone?   
510                                  Hot Girls Wanted   
511  The Other One: The Long Strange Trip of Bob Weir   
512                                        My Own Man   

                          director                                 country  
506              Evgeny Afineevsky  United Kingdom, Ukraine, United States  
507                 Morgan Neville                           United States  
508  Kristina Goolsby, Ashley York                           United States  
509                     Liz Garbus                           United States  
510       Jill Bauer, Ronna Gradus                           United States  
511                    Mike Fleiss                           United States  
512                David Sampliner                           United States

In [40]:
df_concat.iloc[15:26, 0:5]


show_id   type                              title  \
15    s960  Movie                       The Disciple   
16    s965  Movie                Things Heard & Seen   
17    s967  Movie                      Get the Grift   
18   s1024  Movie                    Ajeeb Daastaans   
19   s1025  Movie             Arlo the Alligator Boy   
20   s1051  Movie                        Ride or Die   
21   s1071  Movie               Why Did You Kill Me?   
22   s1086  Movie                  Night in Paradise   
23   s1089  Movie                      Thunder Force   
24   s1093  Movie  Dolly Parton: A MusiCares Tribute   
25   s1106  Movie                    Concrete Cowboy   

                                             director  \
15                                  Chaitanya Tamhane   
16              Robert Pulcini, Shari Springer Berman   
17                                      Pedro Antonio   
18  Shashank Khaitan, Raj Mehta, Neeraj Ghaywan, K...   
19                                         Ryan Crego   
20                                     Ryuichi Hiroki   
21                                      Fredrick Munk   
22                                     Park Hoon-jung   
23                                        Ben Falcone   
24                                                NaN   
25                                        Ricky Staub   

                                                 cast  
15  Aditya Modak, Arun Dravid, Sumitra Bhave, Deep...  
16  Amanda Seyfried, James Norton, Natalia Dyer, A...  
17  Marcus Majella, Samantha Schmütz, Caito Mainie...  
18  Fatima Sana Shaikh, Jaideep Ahlawat, Armaan Ra...  
19  Michael J. Woodard, Mary Lambert, Brett Gelman...  
20  Kiko Mizuhara, Honami Sato, Yoko Maki, Anne Su...  
21                                                NaN  
22  Um Tae-goo, Jeon Yeo-been, Cha Seoung-won, Lee...  
23  Melissa McCarthy, Octavia Spencer, Jason Batem...  
24                                       Dolly Parton  
25  Idris Elba, Caleb McLaughlin, Jharrel Jerome, ...

In [41]:
df_concat.iloc[20:31, [2,7, 8]]


title  release_year rating
20                        Ride or Die          2021  TV-MA
21               Why Did You Kill Me?          2021  TV-MA
22                  Night in Paradise          2021  TV-MA
23                      Thunder Force          2021  PG-13
24  Dolly Parton: A MusiCares Tribute          2021  TV-PG
25                    Concrete Cowboy          2021      R
26                       Just Say Yes          2021  TV-MA
27                      Madame Claude          2021  TV-MA
28               Tersanjung the Movie          2021  TV-14
29                        A Week Away          2021  TV-PG
30                           Bad Trip          2021  TV-MA